provider/docker/docker.go

```diff
@@ -721,6 +725,45 @@ func (p *Provider) getBasicAuth(container dockerData) []string {
 	return []string{}
 }
 
+func (p *Provider) hasRequestHeaders(container dockerData) bool {
+	if _, err := getLabel(container, types.LabelFrontendRequestHeader); err != nil {
+		return false
+	}
+	return true
+}
+
+func (p *Provider) hasResponseHeaders(container dockerData) bool {
+	if _, err := getLabel(container, types.LabelFrontendResponseHeader); err != nil {
+		return false
+	}
+	return true
+}
+
+func (p *Provider) getRequestHeaders(container dockerData) map[string]string {
+	return parseCustomHeaders(container, types.LabelFrontendRequestHeader)
+}
+
+func (p *Provider) getResponseHeaders(container dockerData) map[string]string {
+	return parseCustomHeaders(container, types.LabelFrontendResponseHeader)
+}
+
+func parseCustomHeaders(container dockerData, containerType string) map[string]string {
+	h := make(map[string]string)
+	if r, err := getLabel(container, containerType); err == nil {
```

Comment:
maybe you could use explicit variable name instead of one letter.

By example:
- `h` -> `headers`
- `r` -> `label`
- `v` -> `pairs`
- `k` ->  `pair`